required_scripts/sigprofiler_fdfe4e8.py

```diff
@@ -0,0 +1,100 @@
+                        help='Path to directory containing vcf file(s) - Can be relative or full path')
+
+    parser.add_argument('-g', '--genome', default='GRCh38',
+                        help='Optional definition of genome, defaults to GRCh38')
+
+    parser.add_argument('-p', '--project', metavar="<arg>",
+                        help='Name of the project')
+
+    parser.add_argument('-e', '--exome', default=False, action="store_true",
+                        help='Set if input is from exome')
+
+    parser.add_argument('-t', '--threads', default=-1, type=int,
+                        help='Set number of threads, default will use all threads')
+
+    # prints help message when 0 arguments are entered
+    if len(sys.argv) == 1:
+        parser.print_help(sys.stderr)
+        sys.exit(1)
+
+    parser_args = parser.parse_args()
+
+    return parser_args
+
+def extractSignatures(output,
+                      vcfpath,
+                      genome,
+                      project,
+                      sigClass='SBS',
+                      sigContext='96',
+                      exome=False,
+                      threads=-1):
+    """Extracts signatures from a matrix generated by SigProfilerMatrixGenerator"""
+
+    sigType = sigClass + sigContext
+
+    if exome:
+        input_data = vcfpath+"/output/"+sigClass+"/"+project+"."+sigType+".exome"
+    else:
+        input_data = vcfpath+"/output/"+sigClass+"/"+project+"."+sigType+".all"
+
+    sig.sigProfilerExtractor("matrix", output, input_data, reference_genome=genome, opportunity_genome = genome, context_type=sigType ,minimum_signatures=1, maximum_signatures=10, cpu=threads)
+
+    from SigProfilerExtractor import decomposition as decomp
+    signatures = output+"/"+sigType+"/Suggested_Solution/"+sigType+"_De-Novo_Solution/Signatures/"+sigType+"_De-Novo_Signatures.txt"
+    activities = output+"/"+sigType+"/Suggested_Solution/"+sigType+"_De-Novo_Solution/Activities/"+sigType+"_De-Novo_Activities_refit.txt"
+    samples = output+"/"+sigType+"/Samples.txt"
+
+    #to get all cosmic signatures without filtering
+    decomp.decompose(signatures, activities, samples, output, genome_build=genome, verbose=False, nnls_add_penalty=0.0, nnls_remove_penalty=0.0, initial_remove_penalty=0.0, de_novo_fit_penalty=0.02)
+
+def main():
+    # Parse and validate arguments
+    args = parse_arguments()
+
+    matrices = matGen.SigProfilerMatrixGeneratorFunc(args.project, args.genome, args.vcfpath, plot=True, exome=args.exome, bed_file=None, chrom_based=False, tsb_stat=False, seqInfo=False, cushion=100)
+
+    num_tasks = 0
+    sig_list = []
+
+    if '96' in matrices:
+        num_tasks = num_tasks + 1
+        sig_list.append(('SBS', '96'))
+
+    if 'DINUC' in matrices:
+        num_tasks = num_tasks + 1
+        sig_list.append(('DBS', '78'))
+
+    if 'ID' in matrices:
+        num_tasks = num_tasks + 1
+        sig_list.append(('ID', '83'))
+
+    cpus_per_task = int(args.threads/num_tasks)
```

Comment:
suggestion: what if the value is -1; maybe add a min() function?

Reply:
Fixed in commit 92f3ba3